tests/Routing/RouteSignatureParametersTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Illuminate\Tests\Routing;
+
+use ReflectionParameter;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use PHPUnit\Framework\TestCase;
+use Illuminate\Routing\RouteSignatureParameters;
+use Symfony\Component\HttpFoundation\Request as SymfonyRequest;
+
+if (! class_exists(RouteSignatureParametersTestController::class)) {
```

Comment:
Why do you need all this?

Reply:
Are you referring to the controller mock or the test? I could put the mock in a separate file if you prefer. Regarding the test cases, `RouteSignatureParameters` was untested and crashed on invokable objects so I did my best to deal with that.